sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -0,0 +1,761 @@
+                    switch (type) {
+                        case TypeTags.INT_TAG:
+                            if (value instanceof BigDecimal) {
+                                struct.put(fieldName, ((BigDecimal) value).intValue());
+                            } else {
+                                struct.put(fieldName, value);
+                            }
+                            break;
+                        case TypeTags.FLOAT_TAG:
+                            if (value instanceof BigDecimal) {
+                                struct.put(fieldName, ((BigDecimal) value).doubleValue());
+                            } else {
+                                struct.put(fieldName, value);
+                            }
+                            break;
+                        case TypeTags.DECIMAL_TAG:
+                            if (value instanceof BigDecimal) {
+                                struct.put(fieldName, value);
+                            } else {
+                                struct.put(fieldName, ValueCreator.createDecimalValue((BigDecimal) value));
+                            }
+                            break;
+                        case TypeTags.STRING_TAG:
+                            struct.put(fieldName, value);
+                            break;
+                        case TypeTags.BOOLEAN_TAG:
+                            struct.put(fieldName, ((int) value) == 1);
+                            break;
+                        case TypeTags.OBJECT_TYPE_TAG:
+                        case TypeTags.RECORD_TYPE_TAG:
+                            struct.put(fieldName,
+                                    createUserDefinedType((Struct) value,
+                                            (StructureType) internalField.getFieldType()));
+                            break;
+                        default:
+                            createUserDefinedTypeSubtype(internalField, structType);
+                    }
+                    ++index;
+                }
+            }
+        } catch (SQLException e) {
+            throw new ApplicationError("Error while retrieving data to create " + structType.getName()
+                    + " record. ", e);
+        }
+        return struct;
+    }
+
+    @Override
+    protected void createUserDefinedTypeSubtype(Field internalField, StructureType structType)
+            throws ApplicationError {
+        throw new ApplicationError("Error while retrieving data for unsupported type " +
+                internalField.getFieldType().getName() + " to create "
+                + structType.getName() + " record.");
+    }
+
+    protected static Object[] validateNullable(Object[] objects) {
+        Object[] returnResult = new Object[2];
+        boolean foundNull = false;
+        Object nonNullObject = null;
+        for (Object object : objects) {
+            if (object != null) {
+                if (nonNullObject == null) {
```

Comment:
What is the purpose of this `if` condition, since `nonNullObject` is assigned `null` and has not changed.